src/Data/Vec/Base.agda

```diff
@@ -345,12 +345,12 @@ initLast {n = suc n} (x ∷ xs) with initLast xs
 ... | (ys , y , refl) = (x ∷ ys , y , refl)
 
 init : Vec A (1 + n) → Vec A n
-init xs with initLast xs
-... | (ys , y , refl) = ys
+init {n = zero}  (x ∷ []) = []
+init {n = suc n} (x ∷ xs) = x ∷ init xs
```

Comment:
Do we actually need to match on the implicit `n` here? Can we get away with?
```
init (x ∷ []) = []
init (x ∷ y :: xs) = x ∷ init (y :: xs)
```
Same with `last` below?

Reply:
It turns out that matching on `xs` instead of matching on `n` blocks expressions like `init (y ∷ (xs ∷ʳ x))` from computing. Instead, I can change `init` to signify that `xs` is not actually matched:

```agda
init : Vec A (1 + n) → Vec A n
init {n = zero}  (x ∷ xs) = []
init {n = suc n} (x ∷ xs) = x ∷ init xs
```

The same problem applies to `last` and `last-∷ʳ`.

Reply:
Okay, I edited both `init` and `last` to emphasis that `xs` in the tail is not matched.